pkg/controller/resourceactuation/resourceactuation.go

```diff
@@ -15,14 +15,86 @@
 package resourceactuation
 
 import (
+	"context"
 	"fmt"
+	"strings"
 
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/reconciliationinterval"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
 
+	opv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/operator/pkg/apis/core/v1beta1"
+	opk8s "github.com/GoogleCloudPlatform/k8s-config-connector/operator/pkg/k8s"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// todo acpana: revisit
+var DefaultActuationAction = ActuationAction("reconciling")
+
+type ActuationAction string
+
+func (aa ActuationAction) String() string {
+	return strings.ToLower(string(aa))
+}
+
+func defaultOrLower(s string) string {
+	if s == "" {
+		return DefaultActuationAction.String()
+	}
+
+	return strings.ToLower(s)
+}
+
+func ReconcilingByDefault() bool {
+	return DefaultActuationAction == "reconciling"
+}
+
+// ShouldActuate checks if the resource should be actuated. It errors out if it cannot fetch the
+// ConfigConnector (CC) resource or the ConfigConnectorContext (CCC) resource for the resource's namespace if running
+// in namespaced mode.
+// - If both CC & CCC specify an actuationAction in namespaced mode, we defer to the CCC's value.
+//   - If only CC specifies an actuationAction in namespaced mode, we defer to the CC's value.
+// - If both CC & CCC specify an actuationAction in cluster mode, the CCC specification is irrelevant.
+func ShouldActuate(ctx context.Context, c client.Client, resourceNN types.NamespacedName) (bool, error) {
```

Comment:
could probably further break this up into a helper to fetch the CC, CCC instances and one to decide on the actuation value but leaving it as is for now unless someone objects

Reply:
No objection to only doing the refactoring when needed!